website/content/en/preview/concepts/nodeclasses.md

```diff
@@ -449,14 +454,36 @@ amiSelectorTerms:
   - id: ami-123
 ```
 
-This field is optional, and Karpenter will use the latest EKS-optimized AMIs for the AMIFamily if no amiSelectorTerms are specified. To select an AMI by name, use the `name` field in the selector term. To select an AMI by id, use the `id` field in the selector term. To ensure that AMIs are owned by the expected owner, use the `owner` field - you can use a combination of account aliases (e.g. `self` `amazon`, `your-aws-account-name`) and account IDs.
+To select the latest EKS optimized AMIs, use the `eksOptimized` field in the selector term. To select an AMI by name, use the `name` field in the selector term. To select an AMI by id, use the `id` field in the selector term. To ensure that AMIs are owned by the expected owner, use the `owner` field - you can use a combination of account aliases (e.g. `self`, `amazon`, `your-aws-account-name`) and account IDs.
 
 If owner is not set for `name`, it defaults to `self,amazon`, preventing Karpenter from inadvertently selecting an AMI that is owned by a different account. Tags don't require an owner as tags can only be discovered by the user who created them.
 
 {{% alert title="Tip" color="secondary" %}}
 AMIs may be specified by any AWS tag, including `Name`. Selecting by tag or by name using wildcards (`*`) is supported.
 {{% /alert %}}
 
+{{% alert title="Note" color="primary" %}}
+The `eksOptimized` term is mutually exclusive. When an `eksOptimized` term is specified, other requirements may not be added to the term and no other terms may be specified. The following examples would be invalid:
+
+```yaml
+  amiSelectorTerms:
+    - eksOptimzed:
+        family: AL2023
+      # Can not specify other fields with eksOptimzed
+      tags:
+        karpenter.sh/discover: "${CLUSTER_NAME}"
+        environment: test
+```
+
+```yaml
+  amiSelectorTerms:
+    - eksOptimzed:
+      family: AL2023
+    # Can not specify other terms with eksOptimzed
+    - id: ami-123
```

Comment:
Discussed offline, agreed to tentatively keep this validation on the CRD / webhook level but to not make this assumption in code. Even though we can allow this from a technical standpoint, the latest AMI from any given term would be used as normal, since we know that users are not in control of the release cycle of EKS optimized AMIs mixing this term with others could result in unexpected drift. We can change this in the future if good use-case arguments are made, but for now we'll choose to block a potential footgun. 

The reason we're just doing this on the validation side and not treating this as an assumed invariant in code is there are users who will be able to circumvent this validation. The main case I'm considering is users on 1.23 and 1.24 (no CEL) who haven't deployed the validation webhook, but we can also consider users who edited their CRD. In both of these cases, I believe amiSelectorTerms should work consistently with it's definition, amis resolved from different terms being ored together and the latest compatible AMI being chosen.

Reply:
> The reason we're just doing this on the validation side and not treating this as an assumed invariant in code is there are users who will be able to circumvent this validation

IMO, we should assume all of the validations happen and continue to assume these invariants in code. Yes, it's technically possible that users disable webhooks and are on an early version of K8s, but we shouldn't build our code around these users. It creates too much sprawl; if we are doing something that assumes this isn't the case in code, I'd argue that we should make changes to ensure our code is simpler by assuming this invariant (a good example of us doing this in other ways is assuming that the NodeClassReference is always set)

Reply:
I'd agree but I don't think there's any impact on code complexity whether we consider the invariant or not. The difference is returning one of two slices or appending them together. Since there's no downside from a code complexity / maintainability perspective I think it's really just a question of what is the better UX for users that circumvent the CRD validation, whether that's intentionally or not. IMO it makes more sense to stay consistent with the defined semantics for `amiSelectorTerms` than only include the `eksOptimized` term or all other terms ored together.